internal/bminventory/inventory.go

```diff
@@ -2048,8 +2048,8 @@ func (b *bareMetalInventory) GetClusterSupportedPlatformsInternal(
 		return nil, fmt.Errorf("error getting cluster, error: %w", err)
 	}
 	// no hosts or SNO
-	if len(cluster.Hosts) == 0 || swag.StringValue(cluster.HighAvailabilityMode) != models.ClusterHighAvailabilityModeFull {
-		return &[]models.PlatformType{models.PlatformTypeBaremetal}, nil
+	if cluster.EnabledHostCount <= 1 {
```

Comment:
Is this what we want?  If the user selected a full HA cluster but only has 1 host discovered so far, it won't let him choose bare-metal?

Reply:
TBH I am not sure how SNO should be detected. Is it enough to just check 'HighAvailabilityMode' is 'None'? I mean is 'HighAvailabilityModeNone' exclusive of SNO deployments? Or do we need to check also the number of enabled hosts?

Reply:
isn't that defined by https://github.com/openshift/assisted-service/blob/11413502e6ae30d4b146473bd6a44220398a7f33/swagger.yaml#L7039 ?

Reply:
I don't know how **user_managed_networking** is related to SNO but I found that the cluster manager checks SNO with:

https://github.com/openshift/assisted-service/blob/11413502e6ae30d4b146473bd6a44220398a7f33/internal/cluster/cluster.go#L1154

https://github.com/openshift/assisted-service/blob/11413502e6ae30d4b146473bd6a44220398a7f33/internal/common/common.go#L106-L108

Should I use that function instead of counting enabled hosts?

Reply:
It will probably be better to reuse code